tools/configure/cmdArgs.ml

```diff
@@ -0,0 +1,199 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Util
+
+(** * Command-line parsing *)
+
+type ide = Opt | Byte | No
+
+type nativecompiler = NativeYes | NativeNo | NativeOndemand
+
+module Prefs = struct
+
+type t = {
+  prefix : string option;
+  interactive : bool;
+  libdir : string option;
+  configdir : string option;
+  datadir : string option;
+  mandir : string option;
+  docdir : string option;
+  arch : string option;
+  natdynlink : bool;
+  coqide : ide option;
+  macintegration : bool;
+  browser : string option;
+  withdoc : bool;
+  byteonly : bool;
+  bin_annot : bool;
+  annot : bool;
+  bytecodecompiler : bool;
+  nativecompiler : nativecompiler;
+  coqwebsite : string;
+  warn_error : bool;
+  dune_profile : string;
+  install_enabled : bool;
+}
+
+end
+
+open Prefs
+
+module Profiles = struct
+
+let default = {
+  prefix = None;
+  interactive = true;
+  libdir = None;
+  configdir = None;
+  datadir = None;
+  mandir = None;
+  docdir = None;
+  arch = None;
+  natdynlink = true;
+  coqide = None;
+  macintegration = true;
+  browser = None;
+  withdoc = false;
+  byteonly = false;
+  bin_annot = false;
+  annot = false;
+  bytecodecompiler = true;
+  nativecompiler =
+    if os_type_win32 || os_type_cygwin then NativeNo else NativeOndemand;
+  coqwebsite = "http://coq.inria.fr/";
+  warn_error = false;
+  dune_profile = "release";
+  install_enabled = true;
+}
+
+let devel state = { state with
+  bin_annot = true;
+  annot = true;
+  warn_error = true;
+  dune_profile = "dev";
+  interactive = true;
+  prefix = Some (Filename.concat (Sys.getcwd ()) "_build_vo/default");
+  install_enabled = false;
+}
+
+let devel_doc = "-annot -bin-annot -warn-error yes"
+
+let get = function
+  | "devel" -> devel
+  | s -> raise (Arg.Bad ("profile name expected instead of "^s))
+
+let doc =
+  "<profile> Sets a bunch of flags. Supported profiles:
+     devel = " ^ devel_doc
+
+end
+
+let get_bool = function
+  | "true" | "yes" | "y" | "all" -> true
```

Comment:
Do we also want to accept capitalizations? `Y`?

Reply:
I am keeping the current ones for now, no change here [other than moving code]